compaction/compaction.hh

```diff
@@ -9,6 +9,7 @@
 
 #pragma once
 
+#include "readers/combined.hh"
```

Comment:
This is adding a dependency of the whole reader hierarchy. Let's split of the stats struct into its own header file to reduce rebuilds.